### ISSUE TITLE
Add custom OG image for individual summaries

### ISSUE THREAD

**Issue by HarshKapadia2** [OurTechCommunity/catchup#3]
- Generate an OG image for every individual summary.
- Preferably automate it.
- Beware of varying image sizes on different platforms leading to the image text getting cut.
- Eg: https://twitter.com/DeepankarBhade/status/1468929250354860032

**Comment by HarshKapadia2** [comment]
Maybe use https://github.com/vercel/og-image?

**Comment by HarshKapadia2** [comment]
Look into [`html2canvas`](https://www.npmjs.com/package/html2canvas). ([Doesn't seem to be Node.js compatible though.](https://www.npmjs.com/package/html2canvas#:~:text=It%20does%20not%20require%20any%20rendering%20from%20the%20server%2C%20as%20the%20whole%20image%20is%20created%20on%20the%20client%27s%20browser.%20However%2C%20as%20it%20is%20heavily%20dependent%20on%20the%20browser%2C%20this%20library%20is%20not%20suitable%20to%20be%20used%20in%20nodejs.))

**Comment by tusharnankani** [comment]
Ohhh nice! @SirusCodes 
Although I guess we should have an OG image like this:
![image](https://user-images.githubusercontent.com/61280281/201480129-6adc2ed7-1250-40a0-b347-cd0583d431d8.png)

https://og-image.vercel.app/OTC%20CatchUp%20103%20Summary.png

Creating something around this looks promising!

**Comment by SirusCodes** [comment]
Yeah but the reason for having a screenshot is that people can have a glimpse of what summary has and can read a few lines and if they want they can open it and continue reading...

**Comment by KartikSoneji** [comment]
I am still considering writing a custom renderer.

**Comment by SirusCodes** [comment]
@KartikSoneji I'm thinking that it is worth the effort or not. I feel it's better to implement it as a standalone project so others can also use which kind of justifies the efforts.

But until that is implemented, I feel it's better to use https://screenshot.as-a-service.dev/

**Comment by tusharnankani** [comment]
> I am still considering writing a custom renderer.

Do you mean writing a complete og-image service or using OG Image as a serivce like Vercel og-image? @KartikSoneji

**Comment by KartikSoneji** [comment]
@tusharnankani sorry I meant a custom asciidoctor renderer for adoc -> svg.

**Comment by tusharnankani** [comment]
Updates? @KartikSoneji

**Comment by tusharnankani** [comment]
Can we have summary screenshot as the OG Image? cc: @PranavDani

**Comment by SirusCodes** [comment]
> Can we have summary screenshot as the OG Image?

I still feel having screenshot is a better

**Comment by KartikSoneji** [comment]
> Updates? @KartikSoneji

Sorry, need to open an issue in Asciidoctor regarding the internal flow of the rendering pipeline.
I had found a workaround but I'm not sure if it's better to fix the underlying issue.

**Comment by KartikSoneji** [comment]
> > Can we have summary screenshot as the OG Image?
> 
> I still feel having screenshot is a better

<video src="https://user-images.githubusercontent.com/25801283/221710559-d1d84118-dbe1-464b-9fd0-01a78eaf415a.mp4" muted autoplay />

**Comment by HarshKapadia2** [comment]
One way to do this: [Social media cards generated with shot-scraper](https://til.simonwillison.net/shot-scraper/social-media-cards)

[Source](https://twitter.com/tusharnankanii/status/1695484334122193281)
